### ISSUE TITLE
Familiars: Add pet tag when a creature becomes a familiar

### ISSUE THREAD

**Issue by wolfbain5** [SoSly/ArcaneAdditions#77]
to add compatibility with other mods, adding the pet tag to a creature after it becomes a familiar so that items and artifacts designed for when players have a pet with them will work.

**Comment by nivthefox** [comment]
The question, here, is: "What is a Pet Tag"?  I would love to do this, but I can't see any default effect for this.  It's probable that I'm missing something, so I'll keep digging.

**Comment by celestmia** [comment]
I think "pet tag" could refer to how you can be the owner of certain animals by taming them (`OwnableEntity` interface). You get recorded as the owner. Some mods may use that owner information on the pet to do various things like prevent you from accidentally hurting them.

This sounds like it'd be a bit challenging to do though, since you can turn non-tamable animals into familiars as well.

**Comment by nivthefox** [comment]
@celestmia Right, this is why I am not sure this is doable. There's really no built-in way to do this short of changing class parent, which is not a thing that can be done.
